bigframes/operations/__init__.py

```diff
@@ -148,6 +145,21 @@ def order_preserving(self) -> bool:
         return False
 
 
+@dataclasses.dataclass(frozen=True)
+class NaryOp:
```

Comment:
Thoughts on making this a superclass to Unary, Binary, Ternary?

Reply:
makes sense, done